src/odemis/odemisd/modelgen.py

```diff
@@ -453,7 +461,15 @@ def _make_args(self, name):
                 else:
                     # the child has a class or is created by another component
                     # => we explicitly reuse it
-                    init["children"][internal_role] = self._get_component_by_name(child_name)
+                    init["dependencies"][internal_role] = self._get_component_by_name(child_name)
+        if "dependencies" in init and class_name == "Microscope":
```

Comment:
This can never happen, as you've already checked for `"dependencies" in init`. Probably you wanted to check for `"dependencies" in attr`?